intutil/math.go

```diff
@@ -98,3 +98,10 @@ func (decrementor *Decrementor) Decrement() int {
 func (decrementor *Decrementor) GetInitialMax() int {
 	return int(decrementor.initialMax)
 }
+
+// Range limits value to a given minimum and maximum
+func Range[T constraints.Ordered](value, min, max T) T {
```

Comment:
`range` usually means "range of values"  rather than "constrain within range" (`range` keyword in golang, python `range(start, stop, step) `, and R `range(x, na.rm = FALSE)`), and in mathematics just means the difference between the min and max. I would suggest we rename this to `Constrain` or `Clamp`.